influxdb/src/client/mod.rs

```diff
@@ -301,12 +301,9 @@ impl Client {
 
 pub(crate) fn check_status(res: &HttpResponse) -> Result<(), Error> {
     let status = res.status();
-    if status == StatusCode::UNAUTHORIZED.as_u16() {
-        Err(Error::AuthorizationError)
-    } else if status == StatusCode::FORBIDDEN.as_u16() {
-        Err(Error::AuthenticationError)
-    } else {
-        Ok(())
+    match status {
+        StatusCode::OK => Ok(()),
```

Comment:
Does influxdb specify that it will only use 200 errors? Other ok-like status codes like _204 No Content_ are also commonly used by apis, so we should probably check for non-2xx and not non-200 here?

Reply:
The InfluxDB spec for the response codes is here: https://docs.influxdata.com/influxdb/v2/api/#tag/Response-codes It can return 200 ans 204 for success and a host of other response code in other situations. This can definitely be handled better.

Reply:
I have handled all 2xx responses as successful via comparing the first byte